File: src/main/java/me/nobaboy/nobaaddons/features/chatcommands/GuildCommands.java
package me.nobaboy.nobaaddons.features.chatcommands;

import me.nobaboy.nobaaddons.NobaAddons;
import me.nobaboy.nobaaddons.features.chatcommands.impl.shared.HelpCommand;
import me.nobaboy.nobaaddons.features.chatcommands.impl.shared.WarpOutCommand;
import net.minecraft.util.StringUtils;
import net.minecraftforge.client.event.ClientChatReceivedEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

import java.util.Optional;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class GuildCommands extends ChatCommandManager {
    // .*? is for the player symbols added by mods
    private static final Pattern chatPattern = Pattern.compile("^Guild > .*?(?:\\[[A-Z+]+] )?(?<username>[A-z0-9_]+)(?<grank> \\[[A-z0-9 ]+])?.*?: !(?<command>[A-z0-9_]+) ?(?<argument>[A-z0-9_]+)?");

    {
        register(new HelpCommand(this, "gc", () -> NobaAddons.config.guildHelpCommand));
        register(new WarpOutCommand("gc", () -> NobaAddons.config.guildWarpOutCommand));
    }

    @Override
    protected Optional<Matcher> matchMessage(String message) {
        Matcher match = chatPattern.matcher(message);
        if(match.find()) {
            return Optional.of(match);
        }
        return Optional.empty();
    }


    @SubscribeEvent
    public void onChatReceived(final ClientChatReceivedEvent event) {
        if(!NobaAddons.config.guildCommands) return;
        String receivedMessage = StringUtils.stripControlCodes(event.message.getUnformattedText());

        if(WarpOutCommand.isWarpingOut) {
            if(receivedMessage.toLowerCase().contains(WarpOutCommand.player + " is already in the party")) {
                WarpOutCommand.isWarpingOut = false;
                return;
            } else if(receivedMessage.toLowerCase().contains(WarpOutCommand.player + " joined the party")) {
                WarpOutCommand.playerJoined = true;
                return;
            }
        }

        processMessage(receivedMessage);
    }
}

File: src/main/kotlin/me/nobaboy/nobaaddons/features/chatcommands/impl/GuildCommands.kt
package me.nobaboy.nobaaddons.features.chatcommands.impl

import me.nobaboy.nobaaddons.NobaAddons
import me.nobaboy.nobaaddons.features.chatcommands.ChatCommandManager
import me.nobaboy.nobaaddons.features.chatcommands.impl.shared.HelpCommand
import me.nobaboy.nobaaddons.features.chatcommands.impl.shared.WarpOutCommand
import me.nobaboy.nobaaddons.util.StringUtils.cleanMessage
import me.nobaboy.nobaaddons.util.StringUtils.lowercaseContains
import me.nobaboy.nobaaddons.util.StringUtils.matchMatcher
import net.minecraftforge.client.event.ClientChatReceivedEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import java.util.regex.Matcher
import java.util.regex.Pattern

class GuildCommands : ChatCommandManager() {
    private val chatPattern: Pattern =
        Pattern.compile("^Guild > .*?(?:\\[[A-Z+]+] )?(?<username>[A-z0-9_]+)(?<grank> \\[[A-z0-9 ]+])?.*?: [!?.](?<command>[A-z0-9_]+) ?(?<argument>[A-z0-9_]+)?")

    init {
        register(HelpCommand(this, "gc", NobaAddons.config.guildHelpCommand))
        register(WarpOutCommand("gc", NobaAddons.config.guildWarpOutCommand))
    }

    override fun matchMessage(message: String): Matcher? {
        chatPattern.matchMatcher(message) {
            return this
        }
        return null
    }

    @SubscribeEvent
    fun onChatReceived(event: ClientChatReceivedEvent) {
        if (!isEnabled()) return

        val receivedMessage = event.message.unformattedText.cleanMessage()

        if (WarpOutCommand.isWarpingOut) {
            if (receivedMessage.lowercaseContains("${WarpOutCommand.player} is already in the party")) {
                WarpOutCommand.isWarpingOut = false
                return
            } else if (receivedMessage.lowercaseContains("${WarpOutCommand.player} joined the party")) {
                WarpOutCommand.playerJoined = true
                return
            }
        }

        processMessage(receivedMessage)
    }

    fun isEnabled() = NobaAddons.config.guildCommands
}